tests/driver/Firestore/InProcessFirestore.bulkWriter.test.ts

```diff
@@ -0,0 +1,254 @@
+            name: 'San Francisco',
+            state: 'CA',
+            country: 'USA',
+            capital: false,
+            population: 1000000,
+        })
+    })
+
+    test('create operation creates the doc correctly', async () => {
+        // When we get a new BulkWriter
+        const bulkWriter = db.bulkWriter()
+        
+        // And create a doc with data
+        const docRef = db
+            .collection('cities')
+            .doc('SF')
+        
+        bulkWriter.create(docRef, {
+            name: 'San Francisco',
+            state: 'CA',
+            country: 'USA',
+            capital: false,
+            population: 860000,
+        })
+
+        // when we flush
+        await bulkWriter.flush()
+
+        // we expect the doc to be created
+        const doc = await docRef.get()
+        expect(doc.data()).toEqual({
+            name: 'San Francisco',
+            state: 'CA',
+            country: 'USA',
+            capital: false,
+            population: 860000,
+        })
+    })
+
+    test('delete operation deletes the doc correctly', async () => {
+        // given an existing doc
+        const docRef = db
+            .collection('cities')
+            .doc('LA')
+        
+        await docRef.set({
+            name: 'Los Angeles',
+            state: 'LA',
+            country: 'USA',
+            capital: false,
+            population: 4000000,
+        })
+
+        // When we get a new BulkWriter
+        const bulkWriter = db.bulkWriter()
+
+        // And delete the LA doc
+        bulkWriter.delete(docRef)
+
+        // when we flush
+        await bulkWriter.flush()
+
+        // we expect the doc to have been deleted
+        const doc = await docRef.get()
+        expect(doc.exists).toBeFalsy()
+        expect(doc.data()).toEqual({})
+    })
+    test('cannot create existing document in a bulk write', async () => {
+        // Given there is an existing document
+        await db
+            .collection('animals')
+            .doc('tiger')
+            .set({ description: 'stripey' })
+
+        // When we create the same document
+        let error: Error | null = null
+        const bulkWriter = db.bulkWriter()
+        
+        bulkWriter.create(db.doc('/animals/tiger'), {
+            size: 'large',
+        }).catch((createError) => {
+            error = createError
+        })
+
+        await bulkWriter.flush()
+ 
+        // Then the write should fail
+        expect(error).isFirestoreErrorWithCode(
+            GRPCStatusCode.ALREADY_EXISTS,
+            new RegExp('animals/tiger'),
+        )
+
+        // And the document should not be changed.
+        const snapshot = await db
+            .collection('animals')
+            .doc('tiger')
+            .get()
+        expect(snapshot.exists).toBeTruthy()
+        expect(snapshot.data()).toEqual({ description: 'stripey' })
+    })
+
+    test('throws if an undefined field is written', async () => {
```

Comment:
Can you add a test with two or more writes where one fails and the one succeeds?